### PR TITLE
mwan3: enhancement add ping size and flush conntrack table

### PR DESCRIPTION
**Pull request by feckert**
Maintainer: me
Compile tested: LEDE master
Run tested: lantiq, not upstream, LEDE master, tests done but more test welcome :-)

Description:

I am using mwan3 in an fail over scenario, and i am on the backup interface (3g/LTE) and mwan3 switches back to the default interface (DSL) because it is back again. Connection which are established during the fail over time still use the backup interface even though the main interface is back.
If i delete the conntrack table the connection will use the main interface again.

Signed-off-by: Florian Eckert Eckert.Florian@googlemail.com

### CODE REVIEW

**Comment by champtar** [comment]
If i have 3 connections and 1 go down your patch kill all connections!?
Maybe add a config param defaulting to false because this behaviour is only OK in simple failover scenario

**Comment by feckert** [comment]
Yes this is right i have a simple failover scenario.
1. eth0
2. dsl
3. 3g/LTE

i have a default rule

config rule 'default_rule'
    option dest_ip '0.0.0.0/0'
    option proto 'all'
    option sticky '0'
    opiton use_policy 'wan_xdsl_lte'

What do you mean with the config tag defaulting?

**Comment by champtar** [comment]
Hi @feckert 
I mean you should introduce a config option with this patch to enable disable this new behaviour, and make it disable by default

**Comment by jow-** [comment]
You d not need conntrack-tools for that. On OpenWrt or LEDE you can use `echo f > /proc/net/nf_conntrack` instead.

**Comment by champtar** [comment]
@feckert you forgot to bump the version

**Comment by feckert** [comment]
@jow- i will use the file interface and not conntrack tool as us suggest

@champtar i will add a option per interface 'flush_conntrack'
the values are 'up' 'down' 'both' 'none' default behavior is none

- up (flush conntrack on interface up)
- down (flush conntrack on interface down)
- both ( flush conntrack on up and down)
- none (don't flush conntrack)

additionally i will bump the version number. Which one?
my proposal is
PKG_VERSION:=2.0 -> 2.1

**Comment by feckert** [comment]
@champtar @jow- 

Can we merge this now if i update the conflicts? Or do i have to change some things
Let know if i have to do something

**Comment by feckert** [comment]
Hello @Adze1502 

could you please review my pull request? Its now on hold for 1,5 month. Is the code OK for your? Or should I change something?

If it is ok for you i will solve the conflicts.

**Comment by feckert** [comment]
@champtar i have updated my PR.

I also added a option to set the ping size for mwan3track.

File: net/mwan3/files/lib/mwan3/mwan3.sh [hunk 1]
```diff
@@ -6,6 +6,8 @@ IPS="/usr/sbin/ipset"
 IPT4="/usr/sbin/iptables -t mangle -w"
 IPT6="/usr/sbin/ip6tables -t mangle -w"
 LOG="/usr/bin/logger -t mwan3 -p"
+ECHO="/bin/echo"
```
Comment (champtar) [inline net/mwan3/files/lib/mwan3/mwan3.sh]: is there a reason not to just use internal shell 'echo'?

**Comment by champtar** [comment]
@feckert just answer my question and you are good to go
also if you can remove @Adze1502 from the maintainer list (add a commit or in another PR)

Thanks

File: net/mwan3/files/lib/mwan3/mwan3.sh (hunk 1 above)
Comment (feckert) [inline net/mwan3/files/lib/mwan3/mwan3.sh]: No there is no reason. I could also use `echo`

**Comment by champtar** [comment]
Thanks @feckert

**Comment by feckert** [comment]
Thanks for merging @champtar

**Pull request merged**